### PR TITLE
Initial port of nuget package docs

### PR DESCRIPTION
**Pull request by merriemcgaw**
Publishing of the nuget structure specification for control library packages.

### CODE REVIEW

File: Docs/sdk/assemblies-for-multiple-vs-versions.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
+## Shipping Windows Forms design-time assemblies for multiple Visual Studio versions in a single NuGet package
+
+You may find that you need to ship more than one version of your design-time assemblies, each targeting a different version of Visual Studio.  For example, you might have basic design-time support for your controls starting in VS 17.6, but due to the introduction of some new designer extensibility APIs in 17.7 etc. you might be able to provide a better user experience in 17.8 and later VS versions. In that case you would want to ship two sets of design-time assemblies, one for Visual Studio version 17.6 and the other for 17.8 in a single NuGet package.  
+
+This feature to support design-time assemblies targeting multiple Visual Studio versions in a single NuGet package is supported from 17.0 onwards.
+
+#### NuGet package structure supporting design-time assemblies targeting multiple Visual Studio versions:
+
+![NuGet package structure supporting design-time assemblies targeting multiple Visual Studio versions](/docs/sdk/images/multi-vs-versions-nuget-structure.png)
```
Comment (dreddy-work) [inline Docs/sdk/assemblies-for-multiple-vs-versions.md:9]: Paths to links need to be case sensitive. If possible, its better if we change directory names to lowercase.

File: README.md [hunk 2]
```diff
@@ -4,6 +4,7 @@ Before you start developing Custom Controls for the Out-Of-Process WinForms Desi
 
 ## Documents
 * [Using Visual Studio Solution Templates for creating Control Designers/Type Editors for the Out-Of-Process .NET Core WinForms Designer](https://github.com/microsoft/winforms-designer-extensibility/tree/main/Templates/TypeEditor).
+* [Structuring Control Library nuget packages](/docs/sdk/control-library-nuget-package-spec.md)
```
Comment (RussKie) [inline README.md:7]:
```suggestion
* [Structuring Control Library NuGet packages](/docs/sdk/control-library-nuget-package-spec.md)
```

Comment (RussKie) [inline README.md:7]: I'd reword these to be more grammatically correct. E.g., to something like:
```suggestion
* [How to create Control Designers/Type Editors for the out-of-process .NET Windows Forms designer using Visual Studio solution templates](https://github.com/microsoft/winforms-designer-extensibility/tree/main/Templates/TypeEditor).
* [How to structure Control Library NuGet packages](/docs/sdk/control-library-nuget-package-spec.md)
```

Are all these docs specific to the **out-of-process** or **out-of-process .NET** designer? If so, then add this to the `## Documents` header and remove from the link text.

File: README.md [hunk 3]
```diff
@@ -4,6 +4,7 @@ Before you start developing Custom Controls for the Out-Of-Process WinForms Desi
 
 ## Documents
 * [Using Visual Studio Solution Templates for creating Control Designers/Type Editors for the Out-Of-Process .NET Core WinForms Designer](https://github.com/microsoft/winforms-designer-extensibility/tree/main/Templates/TypeEditor).
```
Comment (RussKie) [inline README.md:6]:
```suggestion
* [Using Visual Studio solution templates for creating Control Designers/Type Editors for the out-of-process .NET Windows Forms designer](https://github.com/microsoft/winforms-designer-extensibility/tree/main/Templates/TypeEditor).
```

File: Docs/sdk/assemblies-for-multiple-vs-versions.md (hunk 1 above)
Comment (RussKie) [reply Docs/sdk/assemblies-for-multiple-vs-versions.md:9]: They only need to be case sensitive for non-Windows platforms.

File: Docs/sdk/assemblies-for-multiple-vs-versions.md [hunk 4]
```diff
@@ -0,0 +1,31 @@
+## Shipping Windows Forms design-time assemblies for multiple Visual Studio versions in a single NuGet package
+
+You may find that you need to ship more than one version of your design-time assemblies, each targeting a different version of Visual Studio.  For example, you might have basic design-time support for your controls starting in VS 17.6, but due to the introduction of some new designer extensibility APIs in 17.7 etc. you might be able to provide a better user experience in 17.8 and later VS versions. In that case you would want to ship two sets of design-time assemblies, one for Visual Studio version 17.6 and the other for 17.8 in a single NuGet package.
```
Comment (Tanya-Solyanik) [inline Docs/sdk/assemblies-for-multiple-vs-versions.md:3]: `VS`->`Visual Studio`  in the whole doc

File: Docs/sdk/control-library-nuget-package-spec.md [hunk 5]
```diff
@@ -0,0 +1,79 @@
...
+If `lib\<tfm>\Design\WinForms` exists, we will only load files under that root. If it doesn’t, we will fall back to using the `lib\<tfm>\Design` folder as the root.  
+  * This allows control developers to ship Windows Forms and WPF controls and designers in the same package (without file collisions) if they wish.
+  
+## Scenarios and User Experience
+
+### Scenario 1: A control with no custom designer
+
+For a runtime control that does not have a custom designer (`ProgressBar`, for example) we would simply use the default designer. The control files go under `lib\<tfm>`:
+
+  ![](/docs/sdk/images/1.png)
```
Comment (Tanya-Solyanik) [inline Docs/sdk/control-library-nuget-package-spec.md:39]: I would prefer descriptive names for png files. And please put a good description into the `[]` to be used as an Alt-text

File: Docs/sdk/assemblies-for-multiple-vs-versions.md (hunk 4 above)
Comment (merriemcgaw) [reply Docs/sdk/assemblies-for-multiple-vs-versions.md:3]: Done for all but 2 that didn't make sense

File: Docs/sdk/control-library-nuget-package-spec.md (hunk 5 above)
Comment (merriemcgaw) [reply Docs/sdk/control-library-nuget-package-spec.md:39]: Absolutely right, great catch. I've cleaned it up.

Comment (Tanya-Solyanik) [reply Docs/sdk/control-library-nuget-package-spec.md:39]: Did you push this change?

Comment (merriemcgaw) [reply Docs/sdk/control-library-nuget-package-spec.md]: I just did, good catch.

File: Docs/sdk/assemblies-for-multiple-vs-versions.md (hunk 1 above)
Comment (dreddy-work) [reply Docs/sdk/assemblies-for-multiple-vs-versions.md:9]: Were you able to open the image form link here? on my machine (Windows) forcing me the URL being case sensitive. So `docs `in URL needed to be `Docs`. If we are not changing folder names in this PR, we should make sure URLs case sensitive.

Comment (merriemcgaw) [reply Docs/sdk/assemblies-for-multiple-vs-versions.md:9]: Everything is previewing just fine on my machine with the PR as is. The folders I did not change are the ones in all of Klaus's solutions - that would be a pain. But docs\sdk is all lower case.

Comment (RussKie) [reply Docs/sdk/assemblies-for-multiple-vs-versions.md:9]: > Were you able to open the image form link here? on my machine (Windows) forcing me the URL being case sensitive. So `docs `in URL needed to be `Docs`. If we are not changing folder names in this PR, we should make sure URLs case sensitive.


> Everything is previewing just fine on my machine with the PR as is. The folders I did not change are the ones in all of Klaus's solutions - that would be a pain. But docs\sdk is all lower case.

You can preview the docs here: https://github.com/microsoft/winforms-designer-extensibility/blob/McGawNugetDocs/Docs/sdk/assemblies-for-multiple-vs-versions.md

The image isn't showing up.

In general, I suggest keeping all folders and filenames in the lower case.

Comment (merriemcgaw) [reply Docs/sdk/assemblies-for-multiple-vs-versions.md:9]: That was my intent, I thought I had changed docs to lower case - maybe I didn't push that commit yet, because you're right, I don't see it either

Comment (merriemcgaw) [reply Docs/sdk/assemblies-for-multiple-vs-versions.md]: OK, this is working now.

**Review by dreddy-work — APPROVED** [review]

**Review by Tanya-Solyanik — APPROVED** [review]
